httpcore/_async/base.py

```diff
@@ -43,7 +45,9 @@ class AsyncByteStream:
     """
 
     def __init__(
-        self, iterator: AsyncIterator[bytes] = None, close_func: Any = None
+        self,
+        iterator: Optional[AsyncIterator[bytes]] = None,
+        close_func: Optional[Callable] = None,
```

Comment:
Don't think these `Optional` markers are necessary with the default mypy config? Do you have a stricter mypy config locally?

```suggestion
        iterator: AsyncIterator[bytes] = None,
        close_func: Callable = None,
```

There are a couple of instances of this across this diff, not going to highlight them all, but I don't think we use `strict optional` in HTTPX.

Reply:
We don't, no. Those were not mypy errors but I changed them as it seemed clearer to me. I'll change it back 👍

Reply:
Oh okay. Yeah in general we've been relying on implicit `Optional` as documented here: https://mypy.readthedocs.io/en/stable/command_line.html#none-and-optional-handling